arrow-core/src/main/kotlin/arrow/core/extensions/option/applicative/OptionApplicative.kt

```diff
@@ -70,8 +62,8 @@ fun unit(): Option<Unit> = arrow.core.Option
 @Deprecated(
   "@extension kinded projected functions are deprecated",
   ReplaceWith(
-  "map(arg1)",
-  "arrow.core.map"
+  "fix().map(arg1)",
```

Comment:
fix is also deprecated and will eventually be removed. I think in this case it would be just replace with `map(arg1)` ?

Reply:
Yes, the problem is that there is no `map` method defined for a value of type `Kind<ForOption, A>`, but for `Option<A>`. So, if we want to provide a valid replacement, we need to add `fix` in order to have access to the `map` method.

Having said that, I'm ok with removing `fix,` but the user would have to adjust the proposed replacement a bit to have the code compiling.

Reply:
Yes, that's actually what I've been doing but that might not always lead to a correctly path of deprecating I think. But I'm not sure if it's always possible. For example if a user has done something like `val x = Either.applicative<String>().just(1)` it'll be typed to `EitherOf<String, Int>`.

Now if he called `import arrow.core.extension.functor.map` on it like so `val y = x.map { it + 1 }` it should result in `Either<String, Int>`.

This scenario probably requires a manual rewrite in the assignment of `x` to rewrite it to `val x: Either<String, Int> = 1.right()`. And then he doesn't even need te deprecation tools because now it should've picked the concrete one and the `import arrow.core.extension.functor.map` import should show as unused.
In other cases where we need to switch from the `arrow.core.extensions.*` package to `arrow.core.*` package the deprecation tools should work without `fix()`.

A deprecation path with `fix()` in it would still require a manual rewrite, but it all depends on how the user instantiated the program. Furthermore, people that have written tagless programs today will have to rewrite to concrete ones so deprecation paths for them don't exist. For example, we can't deprecated `Functor#map` or `Applicative#just` with `ReplaceWith`.